src/wp-admin/site-editor.php

```diff
@@ -19,16 +19,19 @@
 	);
 }
 
-if ( ! ( current_theme_supports( 'block-template-parts' ) || wp_is_block_theme() ) ) {
-	wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
-}
-
 $is_template_part        = isset( $_GET['postType'] ) && 'wp_template_part' === sanitize_key( $_GET['postType'] );
 $is_template_part_path   = isset( $_GET['path'] ) && 'wp_template_partall' === sanitize_key( $_GET['path'] );
 $is_template_part_editor = $is_template_part || $is_template_part_path;
-
-if ( ! wp_is_block_theme() && ! $is_template_part_editor ) {
-	wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
+$is_patterns             = isset( $_GET['postType'] ) && 'wp_block' === sanitize_key( $_GET['postType'] );
+$is_patterns_path        = isset( $_GET['path'] ) && 'patterns' === sanitize_key( $_GET['path'] );
+$is_patterns_editor      = $is_patterns || $is_patterns_path;
+
+if ( ! wp_is_block_theme() ) {
+	if ( ! current_theme_supports( 'block-template-parts' ) && $is_template_part_editor ) {
+		wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
+	} elseif ( ! $is_patterns_editor ) {
+		wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
+	}
```

Comment:
Oops, I think this is blocking classic themes with `block-template-parts` support to access the template part editor. I think we can adjust it to:

```suggestion
if ( ! current_theme_supports( 'block-template-parts' ) || ! $is_template_part_editor && ! $is_patterns_editor ) {
		wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
	}
```

Reply:
This fix is almost identical to what I tried locally. With this applied, I can now access all the links as advertised in the PR description.

Reply:
```php
if ( ! current_theme_supports( 'block-template-parts' ) || ! $is_template_part_editor && ! $is_patterns_editor ) {
```

In the case of this conditional statement, I think classic themes that do not support block-templates-parts will not be able to access the Patterns page. I think I was able to control all cases correctly with the update using 879cb3b.